libsplinter/src/circuit/template/rules/mod.rs

```diff
@@ -63,9 +61,21 @@ impl Rules {
             circuit_builder = set_metadata.apply_rule(circuit_builder, template_arguments)?;
         }
 
-        builders.set_create_circuit_builder(circuit_builder);
-        builders.set_service_builders(service_builders);
-        Ok(())
+        let mut services = vec![];
+        for service_builder in service_builders {
+            match service_builder.build() {
+                Ok(service) => services.push(service),
+                Err(err) => {
+                    debug!("Failed to build SplinterService: {}", err);
```

Comment:
The Error returned will likely end up in the log, so logging here would result in log duplication.